batchata/providers/gemini/gemini_provider.py

```diff
@@ -0,0 +1,307 @@
+"""Simple Google Gemini provider with batch processing."""
+
+import os
+import warnings
+from typing import Dict, List, Optional
+
+import google.genai as genai_lib
+
+from ...core.job import Job
+from ...core.job_result import JobResult
+from ...exceptions import BatchSubmissionError, ValidationError
+from ..provider import Provider
+from .models import GEMINI_MODELS
+from .message_prepare import prepare_messages
+from .parse_results import parse_results
+
+# Suppress known Google SDK warning about BATCH_STATE_RUNNING
+# The API returns BATCH_STATE_* but SDK expects JOB_STATE_*
+warnings.filterwarnings('ignore', message='.*is not a valid JobState')
+
+
+class GeminiProvider(Provider):
+    """Google Gemini provider with batch processing support."""
+    
+    MAX_REQUESTS = 10000
+    
+    def __init__(self, auto_register: bool = True):
+        """Initialize with Google API key."""
+        api_key = os.getenv("GOOGLE_API_KEY", "").strip()
+        if not api_key:
+            raise ValueError("GOOGLE_API_KEY environment variable is required")
+        
+        self.client = genai_lib.Client(api_key=api_key)
+        super().__init__()
+        self.models = GEMINI_MODELS
+        self._batches: Dict[str, Dict] = {}
+    
+    def validate_job(self, job: Job) -> None:
+        """Validate job configuration."""
+        if not job:
+            raise ValidationError("Job cannot be None")
+        
+        if not self.supports_model(job.model):
+            raise ValidationError(f"Unsupported model: {job.model}")
+        
+        model_config = self.get_model_config(job.model)
+        
+        if job.file and not model_config.supports_files:
+            raise ValidationError(f"Model '{job.model}' does not support file input")
+        
+        if job.response_model and not model_config.supports_structured_output:
+            raise ValidationError(f"Model '{job.model}' does not support structured output")
+        
+        if job.messages:
+            contents, _ = prepare_messages(job)
```

Comment:
Not a biggie, but we might want to separate preparing the message from this function for purity